### PR TITLE
Support profile versions for automate profiles storage

### PR DESCRIPTION
**Pull request by alexpop**
After `inspec compliance login_automate ...`

```
$ be inspec compliance exec compliance://admin/apache-baseline#2.0.1

Profile: DevSec Apache Baseline (apache-baseline)
Version: 2.0.1
...
```

```
$ be inspec compliance exec admin/apache-baseline

Profile: DevSec Apache Baseline (apache-baseline)
Version: 2.0.2
```

```
$ be inspec exec compliance://admin/apache-baseline#2.0.1

Profile: DevSec Apache Baseline (apache-baseline)
Version: 2.0.1
```

```
$ be inspec exec compliance://admin/apache-baseline

Profile: DevSec Apache Baseline (apache-baseline)
Version: 2.0.2
```

```
$ be inspec compliance download compliance://admin/apache-baseline#2.0.1
Downloading `admin/apache-baseline#2.0.1`
Profile stored to apache-baseline#2.0.1.tar.gz
$ inspec json apache-baseline#2.0.1.tar.gz | jq .version
"2.0.1"
```

### CODE REVIEW

File: lib/bundles/inspec-compliance/api.rb [hunk 1]
```diff
@@ -83,8 +83,13 @@ def self.version(config)
     # verifies that a profile
     def self.exist?(config, profile)
       _msg, profiles = Compliance::API.profiles(config)
+      owner, id, ver = profile_split(profile)
       if !profiles.empty?
-        index = profiles.index { |p| "#{p['owner_id']}/#{p['name']}" == profile }
+        index = profiles.index { |p|
```
Comment (adamleff) [inline lib/bundles/inspec-compliance/api.rb]: Multiline blocks should use `do..end` please :)

Comment (adamleff) [inline lib/bundles/inspec-compliance/api.rb]: We never really use the index except for checking to see if we have a match. Could we refactor this to use `#any?` instead while we're in here?

```ruby
profiles.any? do |p|
  # criteria
end
```

... that should return true if there was a match, false if not.

**Review by adamleff — CHANGES_REQUESTED** [review]
Change looks fine overall. Can we please add some unit tests for `.target_url` and `.profile_split`? I'd like to see both tests for profile names that include and don't include a version string to prevent future regressions.

**Comment by adamleff** [comment]
We also need to update the inspec-compliance README since this is a new enhancement.

**Comment by alexpop** [comment]
Makes sense Adam, appreciate the review. Will return next week and finish this up.

**Comment by alexpop** [comment]
Adam, I added a second commit to the PR that:
* addresses the code refactor you mentioned
* updates the inspec-compliance bundle README
* adds tests (Wasn't able to stub the `profiles` method in the `exist?` test. Can you have a look at that please?)

Cheers!

**Comment by adamleff** [comment]
@alexpop I think I squared away the pending errors and did a little refactor of one method. Please check 33f57aa0 and see if that meets your needs. The tests should go green shortly.

**Review by adamleff — APPROVED** [review]
Looks good with my clean-up commit. @alexpop please confirm!

**Comment by alexpop** [comment]
Looks very good Adam, happy to merge!

**Pull request merged**